lib/obj2gltf.js

```diff
@@ -78,64 +79,71 @@ function obj2gltf(objPath, gltfPath, options) {
         throw new DeveloperError('gltfPath is required');
     }
 
+    if (metallicRoughness + specularGlossiness + materialsCommon > 1) {
+        throw new DeveloperError('Only one material type may be set from [--metallicRoughness, --specularGlossiness, --materialsCommon].');
+    }
+
+    if (defined(options.metallicRoughnessOcclusionTexture) && defined(options.specularGlossinessTexture)) {
+        throw new DeveloperError('options.metallicRoughnessOcclusionTexture and options.specularGlossinessTexture cannot both be defined.');
+    }
+
+    if (defined(options.metallicRoughnessOcclusionTexture)) {
+        options.metallicRoughness = true;
+        options.specularGlossiness = false;
+        options.materialsCommon = false;
+    }
+
+    if (defined(options.specularGlossinessTexture)) {
+        options.metallicRoughness = false;
+        options.specularGlossiness = true;
+        options.materialsCommon = false;
+    }
+
     var extension = path.extname(gltfPath).toLowerCase();
     var modelName = path.basename(gltfPath, path.extname(gltfPath));
-    if (extension === '.glb') {
+    if (binary || extension === '.glb') {
         binary = true;
-    }
-
-    if (binary && bypassPipeline) {
-        return Promise.reject(new RuntimeError('--bypassPipeline does not convert to binary glTF'));
+        extension = '.glb';
     }
 
     gltfPath = path.join(path.dirname(gltfPath), modelName + extension);
-    var resourcesDirectory = options.bypassPipeline ? path.dirname(gltfPath) : obj2gltf._getTempDirectory();
-
-    var aoOptions = ao ? {} : undefined;
-    var kmcOptions = kmc ? {} : undefined;
-
-    var pipelineOptions = {
-        createDirectory : false,
-        basePath : resourcesDirectory,
-        binary : binary,
-        embed : !separate,
-        embedImage : !separateTextures,
-        quantize : compress,
-        compressTextureCoordinates : compress,
-        encodeNormals : compress,
-        preserve : !optimize,
-        optimizeForCesium : optimizeForCesium,
-        smoothNormals : generateNormals,
-        aoOptions : aoOptions,
-        kmcOptions : kmcOptions,
-        textureCompressionOptions : textureCompressionOptions
+
+    var jsonOptions = {
```

Comment:
For a very slight performance bump, we could set this to undefined when outputting `.glb`.